src/Tizen.Applications.WidgetControl/Tizen.Applications/WidgetControl.cs

```diff
@@ -435,6 +435,34 @@ public bool IsNoDisplay
             }
         }
 
+        /// <summary>
+        /// Gets setup app ID of the widget.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <privilege>http://tizen.org/privilege/widget.viewer</privilege>
+        /// <feature>http://tizen.org/feature/shell.appwidget</feature>
+        /// <exception cref="UnauthorizedAccessException">Thrown when an application does not have the required privileges to access this method.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the required features are not supported.</exception>
+        public string SetupAppId
+        {
+            get
+            {
+                string str = Interop.WidgetService.GetSetupAppId(Id);
+                Interop.WidgetService.ErrorCode err =
+                    (Interop.WidgetService.ErrorCode)Internals.Errors.ErrorFacts.GetLastResult();
+                switch (err)
+                {
+                    case Interop.WidgetService.ErrorCode.PermissionDenied:
+                        throw new UnauthorizedAccessException();
+
+                    case Interop.WidgetService.ErrorCode.NotSupported:
+                        throw new NotSupportedException("Not supported");
+                }
```

Comment:
If you put null in the constructor, Id can be null.
At that time, WIDGET_ERROR_INVALID_PARAMETER occurs.

Also, when _get_app_id_of_setup_app is called from widget_service_get_app_id_of_setup_app, other errors appear to occur.

So, it seems that the case for the invalid parameter and the default case must also be handled.

Reply:
exceptions are added